### ISSUE TITLE
Finetunning crashed due to an OSError exception

### ISSUE THREAD

**Issue by jgilibert** [kratzert/finetune_alexnet_with_tensorflow#46]
I get this exception at the trainning part, but it doesn't happen always (it looks a random issue):

`2018-04-18 15:02:15.902533 Start validation
Traceback (most recent call last):
  File "finetune.py", line 188, in <module>
    print("{} Start validation".format(datetime.now()))
OSError: raw write() returned invalid length 90 (should have been between 0 and 45)`

I'm using Windows 10 64 bits with Python 3.5.2 :: Anaconda 4.2.0 (64-bit) and Tensorflow 1.2.1.
My dataset is _Dogs VS Cats_.

Besides, I'd like to recover the already trained model, since it's very upsetting getting this crash after some epoches. As you indicated, I tried to change the _load_initial_weights_ line to this line:
`saver.restore(sess, "/path/to/checkpoint/model.ckpt")`

But I have three _cktp_ files (index, data and meta). What should I do?
Thanks!

**Comment by kratzert** [comment]
I have no clue on your first issue but a quick search suggest it's a windows/python problem. Try using a more recent version of Python and TensorFlow, Update Anaconda etc. This should hopefully fix the problem.

Beside the last question: These 3 files belong together but the weights are stored in the "data" one if I remember correctly. I'm sitting at a freshly setup Laptop and don't have examples available at the moment. Anyway, TensorFlow will handle this for you. Just use the name of the checkpoint files (the part you have set and all files have in common) and add the .ckpt. TensorFlow will automatically load the correct file.
BTW: this could have been resolved by a simple google search and has nothing to do with this code repository. Would appreciate some effort next time before posting here.

**Comment by jgilibert** [comment]
Of course, I googled and tried everything before bothering here. Tensorflow didn't work properly with the last Python or Anaconda version. That's why I had to downgrade to Python 3.5 version.
Regarding the another question, I tried that too, but I got an error (didn't write down, sorry). I came here to ask this since maybe you could help me with this issue for this specific script.

Anyway, I moved to a fresh Debian 9 computer, same version of Tensorflow and Python. The script keeps crashing but with a different issue. I'm getting now this logs after a few minutes in the first epoch:

> `2018-04-19 11:50:41.381104: I tensorflow/core/common_runtime/bfc_allocator.cc:674] 1 Chunks of size 1236736 totalling 1.18MiB
2018-04-19 11:50:41.381111: I tensorflow/core/common_runtime/bfc_allocator.cc:674] 2 Chunks of size 1769472 totalling 3.38MiB
2018-04-19 11:50:41.381118: I tensorflow/core/common_runtime/bfc_allocator.cc:674] 3 Chunks of size 2109440 totalling 6.04MiB
2018-04-19 11:50:41.381126: I tensorflow/core/common_runtime/bfc_allocator.cc:674] 2 Chunks of size 2654208 totalling 5.06MiB
2018-04-19 11:50:41.381133: I tensorflow/core/common_runtime/bfc_allocator.cc:674] 2 Chunks of size 4194304 totalling 8.00MiB
2018-04-19 11:50:41.381141: I tensorflow/core/common_runtime/bfc_allocator.cc:674] 1 Chunks of size 4413952 totalling 4.21MiB
2018-04-19 11:50:41.381148: I tensorflow/core/common_runtime/bfc_allocator.cc:674] 1 Chunks of size 6297344 totalling 6.00MiB
2018-04-19 11:50:41.381156: I tensorflow/core/common_runtime/bfc_allocator.cc:674] 6 Chunks of size 7455232 totalling 42.66MiB
2018-04-19 11:50:41.381164: I tensorflow/core/common_runtime/bfc_allocator.cc:674] 1 Chunks of size 9793536 totalling 9.34MiB
2018-04-19 11:50:41.381172: I tensorflow/core/common_runtime/bfc_allocator.cc:674] 1 Chunks of size 13279232 totalling 12.66MiB
2018-04-19 11:50:41.381180: I tensorflow/core/common_runtime/bfc_allocator.cc:674] 1 Chunks of size 67108864 totalling 64.00MiB
2018-04-19 11:50:41.381188: I tensorflow/core/common_runtime/bfc_allocator.cc:674] 1 Chunks of size 79148544 totalling 75.48MiB
2018-04-19 11:50:41.381196: I tensorflow/core/common_runtime/bfc_allocator.cc:674] 1 Chunks of size 201326592 totalling 192.00MiB
2018-04-19 11:50:41.381203: I tensorflow/core/common_runtime/bfc_allocator.cc:678] Sum Total of in-use chunks: 7.71GiB
2018-04-19 11:50:41.381214: I tensorflow/core/common_runtime/bfc_allocator.cc:680] Stats: 
Limit:                  8285929472
InUse:                  8283155456
MaxInUse:               8285074688
NumAllocs:                  100545
MaxAllocSize:            268435456`

By searching, this looks a memory trouble, but I'm using the same laptop that before (obviously same RAM). I'll try to change the batch size, but it's really strange.

Thank you.

Thank you

**Comment by kratzert** [comment]
This indeed looks like a memory problem. Try reducing the batch size and see if that helps. Are you training on GPU or CPU? Remember that different OS have different RAM consumptions, so it can totally be that with a different OS you get RAM troubles, where before you hadn't. Anyway I neither use Windows nor Debian (Ubuntu 16.04 here) and can't help with any other OS, since I have no resources to test on these systems. But you must definitely have changed something in the scripts, because with the default batch size I set in the scripts, everything runs fine on my 4GB RAM GPU. 

Beside the batch size, it could also be the buffer size of the Dataset (line 68ff in datagenerator.py). You could also decrease this buffersize. At it is, TensorFlow buffers 100 batches on data in RAM. 
So maybe try to decrease this number first.

**Comment by jgilibert** [comment]
I'm training on CPU. Yes, I know different OS have different RAM consumptions, but I don't think Debian consumes much more than Windows... And I don't think so there's much difference between Debian and Ubuntu for this case.

I only changed in the script the learning rate (because I got the already posted issue _NaN_) and, obviously, the paths. I tried a lot of values and I have this value at _0.0001_ currently. 

Which versions of Tensorflow and Python are you using now? Maybe it could help me.

Thank you!
